tests/cnf/core/network/security/tests/nftables-custom-firewall-test.go

```diff
@@ -0,0 +1,505 @@
+			Expect(err).ToNot(HaveOccurred(), outPut, "Failed to create a static route on the worker node")
+		}
+	})
+
+	AfterAll(func() {
+		By("Edit the machineconfiguration cluster to remove NFTables")
+		updateMachineConfigurationCluster(false)
+
+		By("Remove the static route to the external Pod network on each worker node")
+		for _, testPod := range testPodList {
+			outPut, err := setStaticRoute(testPod, "del", hubIPv4Network, routeMap)
+			Expect(err).ToNot(HaveOccurred(), outPut, "Failed to create a static route on the worker node")
+		}
+	})
+
+	Context("custom firewall", func() {
+		AfterEach(func() {
+			By("Define and delete a NFTables custom rule")
+			createMCAndWaitforMCPStable(tsparams.CustomFireWallDelete)
+			err := mco.NewMCBuilder(APIClient, mcNftablesName).Delete()
+			Expect(err).ToNot(HaveOccurred(), "Failed to get the machineConfig")
+
+		})
+
+		It("Verify the creation of a new custom node firewall NFTables table with an ingress rule",
+			reportxml.ID("77412"), func() {
+
+				By("Setup test environment")
+				masterPod := setupRemoteMultiHopTest(ipv4SecurityIPList, hubIPv4ExternalAddresses,
+					ipv4NodeAddrList, workerNodeList,
+					[]string{"hub-pod-worker-0", "hub-pod-worker-1"}, 8888)
+
+				By("Verify ICMP connectivity between the master Pod and the test pods on the workers")
+				err := cmd.ICMPConnectivityCheck(masterPod, ipv4NodeAddrList, interfaceNet1)
+				Expect(err).ToNot(HaveOccurred(), "Failed to ping the worker nodes")
+
+				By("Verify ingress TCP traffic over port 8888 between the master Pod and the test pods on the workers")
+				err = validateTCPTraffic(masterPod, ipv4NodeAddrList, interfaceNet1, containerFrr)
+				Expect(err).ToNot(HaveOccurred(),
+					"Failed to send TCP traffic over port 8888 to the worker nodes")
+
+				By("Verify egress TCP traffic over port 8888 between the test Pod on the worker and the master pod")
+				err = validateTCPTraffic(testPodWorker0, []string{masterPodIPv4Address},
+					interfaceBrEx, "")
+				Expect(err).ToNot(HaveOccurred(),
+					"Failed to send TCP traffic over port 8888 to the worker nodes")
+
+				By("Define and create a NFTables custom rule blocking ingress TCP port 8888")
+				createMCAndWaitforMCPStable(tsparams.CustomFirewallIngressPort8888)
+
+				By("Verify ingress TCP traffic is blocked over port 8888 between the master Pod and the test" +
+					" pods on the workers")
+				err = validateTCPTraffic(masterPod, ipv4NodeAddrList, interfaceNet1, containerFrr)
+				Expect(err).To(HaveOccurred(),
+					"Successfully sent TCP traffic over port 8888 to the worker nodes")
+
+				By("Verify egress TCP traffic over port 8888 between the test Pod on the worker and the master pod")
+				err = validateTCPTraffic(testPodWorker0, []string{masterPodIPv4Address},
+					interfaceBrEx, "")
+				Expect(err).ToNot(HaveOccurred(),
+					"Failed to send TCP traffic over port 8888 to the worker nodes")
+			})
+	})
+})
+
+func updateMachineConfigurationCluster(nftables bool) {
```

Comment:
Pleasae rename the func and the nftables var

Reply:
changed